tests/rest/media/test_url_preview.py

```diff
@@ -193,9 +193,9 @@ def test_cache_returns_correct_type(self) -> None:
         )
 
         # Clear the in-memory cache
-        self.assertIn("http://matrix.org", self.preview_url._cache)
-        self.preview_url._cache.pop("http://matrix.org")
-        self.assertNotIn("http://matrix.org", self.preview_url._cache)
+        self.assertIn("http://matrix.org", self.preview_url._url_previewer._cache)
+        self.preview_url._url_previewer._cache.pop("http://matrix.org")
```

Comment:
I am fine with not refactoring a test just for the beauty of it, it's effort + you may make the test useless on the way without noticing since you also changed the code.